tests/generic_config_updater/test_multiasic_scenarios.py

```diff
@@ -0,0 +1,217 @@
+
+    if output['rc'] or "Patch applied successfully" not in output['stdout']:
+        logger.info("Patching process broken, the error output is {}").format(output['stdout'])
+        pytest_assert(False, "Patching process broken, the error output is {}").format(output['stdout'])
+
+
+def test_check_idf_isolation_apply_patch(duthost):
+    json_patch = IDF_ISOLATION
+    tmpfile = generate_tmpfile(duthost)
+
+    try:
+        print("The current running config is:")
+        print(duthost.shell("show run all", module_ignore_errors=False)['stdout'])
+        logger.info("The current running config is:")
+        logger.info(duthost.shell("show run all", module_ignore_errors=False)['stdout'])
+
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+
+        if output['rc'] or "Patch applied successfully" not in output['stdout']:
+            logger.info("Patching process broken, the error output is {}".format(output['stdout']))
+            pytest_assert(False, "Patching process broken, the error output is {}").format(output['stdout'])
+
+        cmds = 'sonic-db-cli -n asic0 CONFIG_DB hget "BGP_DEVICE_GLOBAL|STATE" idf_isolation_state'
+        expected_value = "isolated_no_export"
+        redis_value = duthost.shell(cmds, module_ignore_errors=False)['stdout']
+        pytest_assert(redis_value == expected_value, "Config IDF ISOLATION failed")
+
+        cmds = 'sonic-db-cli -n asic1 CONFIG_DB hget "BGP_DEVICE_GLOBAL|STATE" "idf_isolation_state"'
+        expected_value = "isolated_withdraw_all"
+        redis_value = duthost.shell(cmds, module_ignore_errors=False)['stdout']
+        pytest_assert(redis_value == expected_value, "Config IDF ISOLATION failed")
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+
+def test_check_idf_unisolation_apply_patch(duthost):
+    json_patch = IDF_UNISOLATION
+    tmpfile = generate_tmpfile(duthost)
+
+    try:
+        print("The current running config is:")
+        print(duthost.shell("show run all", module_ignore_errors=False)['stdout'])
+        logger.info("The current running config is:")
+        logger.info(duthost.shell("show run all", module_ignore_errors=False)['stdout'])
+
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+
+        if output['rc'] or "Patch applied successfully" not in output['stdout']:
+            logger.info("Patching process broken, the error output is {}".format(output['stdout']))
+            pytest_assert(False, "Patching process broken, the error output is {}").format(output['stdout'])
+
+        cmds = 'sonic-db-cli -n asic0 CONFIG_DB hget "BGP_DEVICE_GLOBAL|STATE" idf_isolation_state'
+        expected_value = "unisolated"
+        redis_value = duthost.shell(cmds, module_ignore_errors=False)['stdout']
+        pytest_assert(redis_value == expected_value, "Config IDF ISOLATION failed")
+
+        cmds = 'sonic-db-cli -n asic1 CONFIG_DB hget "BGP_DEVICE_GLOBAL|STATE" idf_isolation_state'
```

Comment:
Could we check 'sudo idf_isolation status' as well? We might have to apply the same config on all asics for that though. May be have separate cases for isolated_no_export and isolated_withdraw_all?